File: Hard/0004 - medianoftwosortedarrays/medianoftwosortedarrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2)
    {
        ArrayList<Integer> results = new ArrayList<>();
        int i = 0;
        int j = 0;
        
        while (i < nums1.length && j < nums2.length)
        {
            if (nums1[i] <= nums2[j])
                results.add(nums1[i++]);
            else
                results.add(nums2[j++]);
        }

        while (i < nums1.length)
            results.add(nums1[i++]);
        while (j < nums2.length)
            results.add(nums2[j++]);

        int size = results.size();
        if (size % 2 == 0)
            return (results.get(size / 2 - 1) + results.get(size / 2)) / 2.0;
        else
            return results.get(size / 2);
    }
}

File: Hard/0004 - medianoftwosortedarrays/medianoftwosortedarrays.cpp
# include <vector>
using namespace std;

class Solution {
    public:
        double findMedianSortedArrays(vector<int>& nums1,
                vector<int>& nums2)
        {
            vector<int> results;
            int i = 0;
            int j = 0;
            
            while (i < nums1.size() && j < nums2.size())
            {
                if (nums1[i] <= nums2[j])
                    results.push_back(nums1[i++]);
                else
                    results.push_back(nums2[j++]);
            }

            while (i < nums1.size())
                results.push_back(nums1[i++]);
            while (j < nums2.size())
                results.push_back(nums2[j++]);

            if (results.size() % 2 == 0)
                return ((results[results.size() / 2 - 1] + results[results.size() / 2]) / 2.0);
            else
                return results[results.size() / 2];
        }
};